### PR TITLE
[third_party] update openthread master

### PR DESCRIPTION
**Pull request by gjc13**

### CODE REVIEW

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Comment by gjc13** [comment]
@jwhui , friendly reminder

**Review by jwhui — APPROVED** [review]

File: src/core/utils/entropy_utils.c [hunk 1]
```diff
@@ -29,15 +29,15 @@
...
 #include <mbedtls/entropy.h>
-#include <openthread/platform/random.h>
+#include <openthread/platform/entropy.h>
 
 int otrMbedtlsEntropyPoll(void *aData, unsigned char *aOutput, size_t aInLen, size_t *aOutLen)
 {
     (void)aData;
 
     int rval = 0;
     // TODO access to random SHOULD be guarded.
```
Comment (jwhui) [inline src/core/utils/entropy_utils.c]: Do we still need this TODO if we rely on mbedtls threading support?

Comment (jwhui) [reply src/core/utils/entropy_utils.c]: @gjc13, thoughts?

Comment (gjc13) [reply src/core/utils/entropy_utils.c]: TODO removed, see:
https://tls.mbed.org/kb/development/thread-safety-and-multi-threading

**Pull request merged**